packages/config/src/ProfileInfo.ts

```diff
@@ -666,6 +788,49 @@ export class ProfileInfo {
         return argValue;
     }
 
+    // _______________________________________________________________________
+    /**
+     * Initialize a zosmf session configuration object with the arguments
+     * from profArgs
+     *
+     * @param profArgs
+     *      An array of profile argument attributes.
+     *
+     * @returns A session that can be used to connect to zosmf;
+     */
+    private static initZosmfSessCfg(profArgs: IProfArgAttrs[]): ISession {
```

Comment:
So far I think ZosmfSession and code that is specific to z/OSMF sessions has lived in the `@zowe/cli` repo. For this `initZosmfSessCfg` method and the `createZosmfSession` method above:
* If they are specific to z/OSMF sessions, do they belong in a Zowe SDK instead of Imperative?
* If they are not specific to z/OSMF sessions, should we rename these methods in Imperative?

The latter seems true to me - that these methods are not specific to z/OSMF sessions. They could be used for any session that connects to a `protocol://host:port`, and supports basic and token auth.

Perhaps this method could be renamed to `initSessCfg` to make it more generic, so that developers could feel free to use it for other types of sessions: CICS, DB2, etc.

Reply:
Because I was working closely with the ZoweExplorer team, I think I had zosmf on the brain. I renamed the functions to createSession and initSessCfg. While createSession just works for zosmf, if a calling app needed additional properties for a different session type, the app could create a thin layer that coerces the session returned by createSession into the session of choice and add the required properties to that session.